core/src/retransmit_stage.rs

```diff
@@ -41,15 +41,29 @@ fn retransmit(
     let r_bank = bank_forks.read().unwrap().working_bank();
     let bank_epoch = r_bank.get_stakers_epoch(r_bank.slot());
     let mut peers_len = 0;
+    let stakes = staking_utils::staked_nodes_at_epoch(&r_bank, bank_epoch);
+    let (peers, stakes_and_index) = cluster_info
+        .read()
+        .unwrap()
+        .sorted_retransmit_peers_and_stakes(stakes.as_ref());
     for packet in &packets.packets {
-        let (my_index, mut peers) = cluster_info.read().unwrap().shuffle_peers_and_index(
-            staking_utils::staked_nodes_at_epoch(&r_bank, bank_epoch).as_ref(),
-            ChaChaRng::from_seed(packet.meta.seed),
-        );
-        peers_len = cmp::max(peers_len, peers.len());
-        peers.remove(my_index);
+        let (my_index, mut shuffled_stakes_and_index) =
+            cluster_info.read().unwrap().shuffle_peers_and_index(
+                &peers,
+                &stakes_and_index,
+                ChaChaRng::from_seed(packet.meta.seed),
+            );
+        peers_len = cmp::max(peers_len, shuffled_stakes_and_index.len());
+        shuffled_stakes_and_index.remove(my_index);
+        // split off the indexes, we don't need the stakes anymore
+        let indexes = shuffled_stakes_and_index
```

Comment:
Extra unnecessary copy though.

Reply:
yeah, I added this in when I realized having it be `(stake, index)` ruined the interface to the turbine functions (and made fixing the turbine simulation a nightmare). 
It made no noticeable impact when I tested it.